pkg/objstore/client/factory.go

```diff
@@ -18,23 +18,25 @@ import (
 var ErrNotFound = errors.New("no valid GCS or S3 configuration supplied")
 
 // NewBucket initializes and returns new object storage clients.
-// TODO(bplotka): Use that in every command.
 func NewBucket(gcsBucket *string, s3Config s3.Config, reg *prometheus.Registry, component string) (objstore.Bucket, func() error, error) {
 	if *gcsBucket != "" {
 		gcsOptions := option.WithUserAgent(fmt.Sprintf("thanos-%s/%s (%s)", component, version.Version, runtime.Version()))
 		gcsClient, err := storage.NewClient(context.Background(), gcsOptions)
 		if err != nil {
 			return nil, nil, errors.Wrap(err, "create GCS client")
 		}
-		return gcs.NewBucket(*gcsBucket, gcsClient.Bucket(*gcsBucket), reg), gcsClient.Close, nil
+		b := gcs.NewBucket(*gcsBucket, gcsClient.Bucket(*gcsBucket), reg)
```

Comment:
Is that necessary? If you feel it is more readable, sure. I think it's not that needed.